tests/leaderboards/utils.py

```diff
@@ -0,0 +1,142 @@
+import json
+import os
+from typing import Any, Dict, Iterator, List, Optional
+
+
+def filter_dict(d: Dict[str, Any], key: str) -> Dict[str, Any]:
+    """
+    Filter a dictionary to include only the keys that contain the specified substring.
+
+    Args:
+        d (Dict): The dictionary to filter.
+        key (str): The substring to filter the dictionary keys by.
+
+    Returns:
+        Dict: A dictionary containing only the keys that contain the specified substring.
+    """
+    return {k: v for k, v in d.items() if key in k}
+
+
+class ConfigParser:
+    """
```

Comment:
I still think it would be good to be homogeneous, and either use a ConfigParser everywhere, or to use nested dicts everywhere - (and I'm in favor of the latter as it's less likely to introduce issues)

Reply:
by everywehre you mean for both expected results and generated results ?